pkg/storage/internalstorage/resource_storage.go

```diff
@@ -274,8 +274,8 @@ func (s *ResourceStorage) List(ctx context.Context, listObject runtime.Object, o
 	return nil
 }
 
-func (s *ResourceStorage) Watch(ctx context.Context, options *metainternalversion.ListOptions) (watch.Interface, error) {
-	return nil, nil
+func (s *ResourceStorage) Watch(_ context.Context, _ *internal.ListOptions) (watch.Interface, error) {
+	return nil, apierrors.NewMethodNotSupported(s.storageGroupResource, "watch")
```

Comment:
By the way 
Do you have any idea about watch in internal- storage-layer ？

Reply:
This will be something we need to discuss and advance recently

My initial idea is to decouple the resource events between apiserver and synchronizer with the help of some message queue components

Reply:
> This will be something we need to discuss and advance recently
> 
> My initial idea is to decouple the resource events between apiserver and synchronizer with the help of some message queue components

We will enlarge member clusters scale next years. I think we may change the storage in future.

Reply:
I want to separate internalstorage from the clusterpedia repo so that we can iterate on internalstorage separately and deliver new features faster

What do you think?